analytic_hr_department_restriction/models/account_analytic_account.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2025 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = "account.analytic.account"
+
+    department_id = fields.Many2one(
+        comodel_name="hr.department",
+        string="Department",
+        domain="['|',('company_id', '=?', company_id),('company_id', '=', False)]",
```

Comment:
Isn't this obtained with the `check_company=True`? Or at least, it should have both.

Reply:
It is not possible, `check_company=True` only works correctly when the `company_id` field in the corresponding model is required, and in the departments (`hr.department`), the `company_id` field is not required.